index.ts

```diff
@@ -0,0 +1,212 @@
+import { interactiveLogin } from "@azure/ms-rest-nodeauth";
+import { ComputeManagementClient, ComputeManagementModels as Models } from "@azure/arm-compute";
+import { NetworkManagementClient, NetworkManagementModels as NetworkModels } from "@azure/arm-network";
+import { RestError, RestResponse, RequestOptionsBase } from "@azure/ms-rest-js";
+
+function getSubscriptionId(): string {
+  const subscriptionId: string | undefined = process.env["AZURE_SUBSCRIPTION_ID"];
+
+  if (!subscriptionId) {
+    console.error("Please set Azure subscription environmental variable");
+    process.exit(1);
+  }
+
+  return subscriptionId!;
+}
+
+async function getAzureClients(
+  subscriptionId: string
+): Promise<{
+  computeClient: ComputeManagementClient;
+  networkClient: NetworkManagementClient;
+}> {
+  const credentials = await interactiveLogin();
```

Comment:
You may wanna use `loginWithServicePrincipalSecret()` as that is what most of the other samples show.

Reply:
I thought about it but I think it's better to have interactive login which enables you checking out this repository and having a working sample instead of complicating it even more.